joern-cli/frontends/gosrc2cpg/src/main/scala/io/joern/gosrc2cpg/astcreation/AstForStatementsCreator.scala

```diff
@@ -100,7 +107,7 @@ trait AstForStatementsCreator { this: AstCreator =>
   private def astForConditionExpression(condStmt: ParserNodeInfo): Ast = {
     condStmt.node match {
       case ParenExpr => astForNode(condStmt.json(ParserKeys.X)).head
-      case _         => Ast()
+      case _         => astsForStatement(condStmt).head
```

Comment:
Please guarantee that `astForStatement(condStmt)` is always non-empty to avoid `NoSuchElementException` here.

Reply:
Yes, will take that precaution